src/API/Discount.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LemonSqueezy\API;
+
+use function array_map;
+
+use LemonSqueezy\Entity\Discount as DiscountEntity;
+
+class Discount extends AbstractApi
+{
+    public function getAllDiscounts(): array
+    {
+        $discounts = $this->get('/discounts');
+
+        return array_map(function ($discount) {
+            $discountEntity = new DiscountEntity($discount->attributes);
+            $discountEntity->id = (int) $discount->id;
+
+            return $discountEntity;
+        }, $discounts->data);
+    }
+
+    public function getStoreDiscounts(int $storeId): array
+    {
+        $discounts = $this->get('/discounts?filter[store_id]=' . $storeId);
+
+        return array_map(function ($discount) {
+            $discountEntity = new DiscountEntity($discount->attributes);
+            $discountEntity->id = (int) $discount->id;
+
+            return $discountEntity;
+        }, $discounts->data);
+    }
+
+    public function getDiscount(int $discountId): DiscountEntity
+    {
+        $discount = $this->get('/discounts/' . $discountId);
+
+        $discountEntity = new DiscountEntity($discount->data->attributes);
+        $discountEntity->id = (int) $discount->data->id;
+
+        return $discountEntity;
+    }
+
+    /**
+     * @param DiscountEntity $discount
+     * @param array<int|string> $variantIds
+     * @return int
+     */
+    public function createDiscount(DiscountEntity $discount, array $variantIds = []): int
```

Comment:
I know this method looks a bit weird. This method should simplify creating discounts and so I decided to pass the possible product variants for the given discount code as a single parameter. 
The entity itself has no attribute yet to cover that. If you have a better idea, I am happy to improve that!

Reply:
I don't create a feature with a Creation Method, so It's an excellent opportunity for me to read and implement this. I'll be getting back to you with the new changes. BTW, I moved your PR to a feature brand to prepare the whole Feature before adding this to the release.